provider/src/cache.ts

```diff
@@ -91,3 +96,61 @@ export const storeRPCResponseInCache = async (
     return false
   }
 }
+
+/**
+ * Contract CACHING - KV STORAGE
+ */
+
+/**
+ * Retrieves a contract's status from KV storage
+ * Returns true if it's an unlock contract, false if it's explicitly not an unlock contract, null if not found
+ */
+export const getContractStatusFromKV = async (
+  env: Env,
+  networkId: string,
+  lockAddress: string
+): Promise<boolean | null> => {
+  if (!env.LOCK_CACHE) {
+    return null
+  }
+
+  try {
+    // Create a unique key combining network ID and address for multi-chain support
+    const key = `${KV_LOCK_PREFIX}${networkId}_${lockAddress.toLowerCase()}`
+    const value = await env.LOCK_CACHE.get(key)
+
+    // If the key doesn't exist, return null
+    if (value === null) {
+      return null
+    }
+
+    // Return true if value is "1", false if value is "0"
+    return value === '1'
```

Comment:
I worry about the risk of confusing "false" with "0"... 
I think it is probably better of this function has explicit return values (not booleans): UNLOCK_PROTOCOL_CONTRACT, OTHER_CONTRACT and "null" when unset.